global-indexes.md

```diff
@@ -0,0 +1,320 @@
+(PARTITION `p0` VALUES LESS THAN (1735660800)
+ PARTITION `p1` VALUES LESS THAN (1738339200)
+ ...)
+```
+
+In the preceding `t` table, the values in the `id` column are unique. To optimize both point queries and range queries, you can define a clustered index in the table creation statement as `PRIMARY KEY(id, ts)` and a global index without the partitioning column as `UNIQUE KEY id(id)`. This way, point queries based on `id` will use the global index `id` and choose a `PointGet` execution plan, while range queries will use the clustered index. The clustered index requires one less table lookup compared to the global index, improving query efficiency.
+
+The modified table structure is as follows:
+
+```sql
+CREATE TABLE `t` (
+  `id` int NOT NULL,
+  `ts` timestamp NOT NULL,
+  `data` varchar(100) DEFAULT NULL,
+  PRIMARY KEY (`id`, `ts`) /*T![clustered_index] CLUSTERED */,
+  UNIQUE KEY `id` (`id`) /*T![global_index] GLOBAL */
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+PARTITION BY RANGE (UNIX_TIMESTAMP(`ts`))
+(PARTITION `p0` VALUES LESS THAN (1735660800),
+ PARTITION `p1` VALUES LESS THAN (1738339200)
+ ...)
+```
+
+This approach optimizes point queries based on `id` while improving the performance of range queries, and also ensures that the table's partitioning columns are effectively utilized in timestamp-based queries.
+
+## Usage
+
+To create a global index, add the `GLOBAL` keyword in the index definition.
+
+> **Note:**
+>
+> Global indexes affect partition management. Executing `DROP`, `TRUNCATE`, or `REORGANIZE PARTITION` operations will trigger updates to the table-level global indexes. This means that these DDL operations only return after the corresponding global index updates are completed, which might increase execution time.
+
+```sql
+CREATE TABLE t1 (
+    col1 INT NOT NULL,
+    col2 DATE NOT NULL,
+    col3 INT NOT NULL,
+    col4 INT NOT NULL,
+    UNIQUE KEY uidx12(col1, col2) GLOBAL,
+    UNIQUE KEY uidx3(col3),
+    KEY idx1(col1) GLOBAL
+)
+PARTITION BY HASH(col3)
+PARTITIONS 4;
+```
+
+In the preceding example, the unique index `uidx12` and the non-unique index `idx1` become global indexes, while `uidx3` remains a regular unique index.
+
+Note that a clustered index cannot be a global index. For example:
+
+```sql
+CREATE TABLE t2 (
+    col1 INT NOT NULL,
+    col2 DATE NOT NULL,
+    PRIMARY KEY (col2) CLUSTERED GLOBAL
+) PARTITION BY HASH(col1) PARTITIONS 5;
+```
+
+```
+ERROR 1503 (HY000): A CLUSTERED INDEX must include all columns in the table's partitioning function
+```
+
+A clustered index cannot simultaneously serve as a global index. This is because if a clustered index were global, the table would no longer be partitioned. The keys of a clustered index are at the partition level, while global indexes operate at the table level, creating a conflict. If you need to set the primary key as a global index, you must explicitly define it as a non-clustered index, for example: 
```

Comment:
```suggestion
A clustered index cannot also serve as a global index. This is because if a clustered index is global, the table would no longer be partitioned. The key of a clustered index is the key for the partition-level row data, while a global index is defined at the table level, creating a conflict. If you need to make the primary key a global index, you must explicitly define it as a non-clustered index. For example:
```